src/array.c

```diff
@@ -96,6 +98,8 @@ enum cc_stat array_new_conf(ArrayConf const * const conf, Array **out)
     ar->mem_alloc  = conf->mem_alloc;
     ar->mem_calloc = conf->mem_calloc;
     ar->mem_free   = conf->mem_free;
+    ar->is_sorted  = false;
+    ar->comp       = NULL;
```

Comment:
In such case, you need to be consistent and explicitly initialize `ar->size`.
I personally do not think this is a good approach.